crates/sui-types/src/passkey_authenticator.rs

```diff
@@ -0,0 +1,191 @@
+    user_signature: Signature,
+    #[serde(skip)]
+    pub bytes: OnceCell<Vec<u8>>,
+}
+
+impl PasskeyAuthenticator {
+    // test only
+    pub fn new(
+        authenticator_data: Vec<u8>,
+        client_data: Vec<u8>,
+        user_signature: Signature,
+    ) -> Self {
+        Self {
+            authenticator_data,
+            client_data,
+            user_signature,
+            bytes: OnceCell::new(),
+        }
+    }
+
+    pub fn get_pk(&self) -> SuiResult<PublicKey> {
+        PublicKey::try_from_bytes(
+            SignatureScheme::Secp256r1,
+            self.user_signature.public_key_bytes(),
+        )
+        .map_err(|_| SuiError::InvalidAuthenticator)
+    }
+}
+
+/// Necessary trait for [struct SenderSignedData].
+impl PartialEq for PasskeyAuthenticator {
+    fn eq(&self, other: &Self) -> bool {
+        self.as_ref() == other.as_ref()
+    }
+}
+
+/// Necessary trait for [struct SenderSignedData].
+impl Eq for PasskeyAuthenticator {}
+
+/// Necessary trait for [struct SenderSignedData].
+impl Hash for PasskeyAuthenticator {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.as_ref().hash(state);
+    }
+}
+
+impl AuthenticatorTrait for PasskeyAuthenticator {
+    fn verify_user_authenticator_epoch(
+        &self,
+        _epoch: EpochId,
+        _max_epoch_upper_bound_delta: Option<u64>,
+    ) -> SuiResult {
+        Ok(())
+    }
+
+    /// Verify an intent message of a transaction with an zk login authenticator.
+    fn verify_claims<T>(
+        &self,
+        intent_msg: &IntentMessage<T>,
+        author: SuiAddress,
+        _aux_verify_data: &VerifyParams,
+        _zklogin_inputs_cache: Arc<VerifiedDigestCache<ZKLoginInputsDigest>>,
+    ) -> SuiResult
+    where
+        T: Serialize,
+    {
+        let client_data_json: CollectedClientData = serde_json::from_slice(&self.client_data)
+            .map_err(|_| SuiError::InvalidSignature {
+                error: "get client data".to_string(),
+            })?;
+        let parsed_challenge =
+            Base64UrlUnpadded::decode_vec(&client_data_json.challenge).map_err(|e| {
+                SuiError::InvalidSignature {
+                    error: e.to_string(),
+                }
+            })?;
+        let mut hasher = DefaultHash::default();
+        hasher.update(&bcs::to_bytes(&intent_msg).expect("Message serialization should not fail"));
+        let digest = hasher.finalize().digest;
+
+        // check client_data.challenge == sha256(intent_msg(tx))
+        if parsed_challenge != digest {
+            return Err(SuiError::InvalidSignature {
+                error: "invalid challenge".to_string(),
+            });
+        };
+
+        // msg = authenticator_data || sha256(client_data)
+        let mut message = self.authenticator_data.clone();
+        let client_data_hash = Sha256::digest(self.client_data.as_slice()).digest;
+        message.extend_from_slice(&client_data_hash);
```

Comment:
ok we're using sha256 here...is there a reason why we need this vs blake2b? Is it a passkey spec thing?

Reply:
its in the passkey spec

https://github.com/sui-foundation/sips/blob/a6f1eaa9ce940e05794973218cab5720d9d970c0/sips/sip-9.md#signature-verification